src/browser-stream-template-renderer.js

```diff
@@ -0,0 +1,54 @@
+/* global ReadableStream */
+
+/**
+ * A custom Readable stream class for rendering a template result to a stream
+ *
+ * @implements TemplateResultRenderer
+ */
+export class StreamTemplateRenderer {
+  /**
+   * Constructor
+   *
+   * @param { TemplateResult } result - a template result returned from call to "html`...`"
+   * @param { TemplateResultProcessor } processor
+   * @returns { ReadableStream }
+   */
+  constructor(result, processor) {
+    if (typeof ReadableStream === 'undefined') {
+      throw Error('ReadableStream not supported on this platform');
+    }
+
+    return new ReadableStream({
+      process: null,
+      start(controller) {
+        const underlyingSource = this;
+        let stack = [result];
+
+        this.process = processor.getProcessor(
+          {
+            push(chunk) {
+              if (chunk === null) {
+                controller.close();
+                return false;
+              }
+
+              controller.enqueue(chunk);
```

Comment:
For this to be read by a user constructed Response I believe the chunk needs to be encoded using [`TextEncoder.encode`](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder/encode) ([Example here](https://jakearchibald.com/2016/streams-ftw/#serving-a-string-slowly)).